### PR TITLE
biomass reagent list

### LINKED ISSUE #2100: SUGG - Add ability, to put roach/spider meat in biomass container
You can add only red meat (monkeys, cows) to biomass container, but roach/spider meat is more accessible for anyone, so it would be good to add ability to use them as biomass for cloning. Maybe, if red meat add 50 biomass, roach meat can add 25?

В качестве биомассы можно использовать только красное мясо (обезьянки, коровки), но мясо паучков и тараканов более доступно для всех, было бы неплохо добавить возможность засовывать тараканье или паучье мясо в контейнер с биомассой, увеличивая общее число биомассы. К примеру, если красное мясо восстанавливает 50 пунктов биомассы, может быть мясо таракана или паука будет восстанавливать 25?

### PR DESCRIPTION
**Pull request by LagannPM**
fixes #2100

### CODE REVIEW

**Comment by NanakoAC** [comment]
Hmm, unsure about this, Neotheology is quite big on human purity, allowing xeno flesh might not gel with them

Will see what clock says

File: code/game/machinery/bioprinter.dm [hunk 1]
```diff
@@ -73,12 +73,13 @@
...
 		return
 	// Meat for biomass.
-	if(!prints_prosthetics && istype(W, /obj/item/weapon/reagent_containers/food/snacks/meat))
-		stored_matter += 50
-		user.drop_item()
-		user << "<span class='info'>\The [src] processes \the [W]. Levels of stored biomass now: [stored_matter]</span>"
-		qdel(W)
-		return
+	if(!prints_prosthetics)
+		if(W.type in BIOMASS_TYPES)
```
Comment (LethalGhost) [inline code/game/machinery/bioprinter.dm]: It's better to use `is_type_in_list`

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -216,9 +216,9 @@
 		else
 			locked = 0
 			user << "System unlocked."
-	else if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/meat))
+	else if(I.type in BIOMASS_TYPES)
```
Comment (LethalGhost) [inline code/game/machinery/cloning.dm]: Can you use `istype` here to include biomass subtypes too?

File: code/__defines/reagents.dm [hunk 3]
```diff
@@ -0,0 +1,2 @@
+//Used to replenish biomass in clonners, bioprinters etc
+#define BIOMASS_TYPES list(/obj/item/weapon/reagent_containers/food/snacks/meat = 50 ,/obj/item/weapon/reagent_containers/food/snacks/roachmeat= 40 ,/obj/item/weapon/reagent_containers/food/snacks/xenomeat = 20)
```
[the remark this answers is not in the document]
Comment (LagannPM) [reply code/__defines/reagents.dm]: Where from i should take those numbers then, uhm ?

File: code/game/machinery/cloning.dm (hunk 2 above)
Comment (LagannPM) [reply code/game/machinery/cloning.dm]: fixed

File: code/__defines/reagents.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (LudwigVonChesterfield) [reply code/__defines/reagents.dm]: Reagent container and making a biomass reagent perhaps

[the remark this answers is not in the document]
Comment (LagannPM) [reply code/__defines/reagents.dm]: perhaps i revisit this after porting bays reagent module

**Pull request merged**